### PR TITLE
Add Azure AI Document Intelligence loader

### PR DESCRIPTION
**Pull request by trducng**

### CODE REVIEW

File: libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+class AzureAIDocumentIntelligenceLoader(BaseReader):
+    """Utilize Azure AI Document Intelligence to parse document
+
+    As of April 24, the supported file formats are: pdf, jpeg/jpg, png, bmp, tiff,
+    heif, docx, xlsx, pptx and html.
+    """
+
+    _dependencies = ["azure-ai-documentintelligence"]
+
+    endpoint: str = Param("Endpoint of Azure AI Document Intelligence")
```
Comment (taprosoft) [inline libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py]: should we add default credential from env var (e.g: AZURE_FORM_RECOGNIZER_KEY, AZURE_FORM_RECOGNIZER_ENDPOINT) for ease of configuration?

File: libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        with open(file_path, "rb") as fi:
+            poller = self.client_.begin_analyze_document(
+                self.model,
+                analyze_request=fi,
+                content_type="application/octet-stream",
+                output_content_format="markdown",
+            )
+            result = poller.result()
+
+        return [Document(content=result.content, metadata=metadata)]
```
Comment (taprosoft) [inline libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py:68]: Even though Markdown output from Azure API directly is nice, we should note that it should be coupled with SemanticChunking to keep table and section structure in place. Reference: https://python.langchain.com/docs/modules/data_connection/document_transformers/markdown_header_metadata/

Comment (taprosoft) [reply libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py:68]: Also, please note that figure image extraction only available in JSON output mode. In markdown mode we still get ```<figure>``` to specify figure region but only text or OCR content embedded inside.
image

File: libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py (hunk 1 above)
Comment (trducng) [reply libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py]: @taprosoft it's possible. Let me add this.

**Comment by lone17** [comment]
Looking at this, I think we need to make loaders configurable in flowsettings.py, what do you think ?

**Comment by trducng** [comment]
> Looking at this, I think we need to make loaders configurable in flowsettings.py, what do you think ?

@lone17 I did thought about this. However, it would means that `kotaemon`-related component will depend on `flowsettings.py`, which is meant for `ktem`. So I think it would be more reasonable to not assume `flowsettings.py` in `kotaemon`. And let the pipeline creator for `ktem` to decide the loader configurations for their pipeline in `flowsettings.py`

**Comment by trducng** [comment]
> Looking at this, I think we need to make loaders configurable in flowsettings.py, what do you think ?

@lone17 I did think about this. However, it would means that `kotaemon`-related component will depend on `flowsettings.py`, which is meant for `ktem`. So I think it would be more reasonable to not assume `flowsettings.py` in `kotaemon`. And let the pipeline creator for `ktem` to decide the loader configurations for their pipeline in `flowsettings.py`

File: libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py (hunk 2 above)
Comment (trducng) [reply libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py:68]: Alrighty, this should be considered as improvement, as we are still working on the best agnostic representation of these information for a file @taprosoft

**Comment by lone17** [comment]
> > Looking at this, I think we need to make loaders configurable in flowsettings.py, what do you think ?
> 
> @lone17 I did think about this. However, it would means that `kotaemon`-related component will depend on `flowsettings.py`, which is meant for `ktem`. So I think it would be more reasonable to not assume `flowsettings.py` in `kotaemon`. And let the pipeline creator for `ktem` to decide the loader configurations for their pipeline in `flowsettings.py`

@trducng why would `kotaemon`'s components depend on `flowsettings.py` ? `kotaemon` defines the reader class and logic, then `flowsettings.py` will be used to configure which readers are available just like LLMs, Embedding models, etc.

**Comment by trducng** [comment]
@lone17 Then what you mention is the same as this one.

> So I think it would be more reasonable to not assume flowsettings.py in kotaemon. And let the pipeline creator for ktem to decide the loader configurations for their pipeline in flowsettings.py

And that will depend on the settings of ktem and pipeline creator, which doesn't really relate to this PR

File: libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py (hunk 2 above)
Comment (taprosoft) [reply libs/kotaemon/kotaemon/loaders/azureai_document_intelligence_loader.py:68]: Sure, let add an issue for this @trducng.

**Review by taprosoft — APPROVED** [review]
LGTM

**Comment by lone17** [comment]
> @lone17 Then what you mention is the same as this one.
> 
> > So I think it would be more reasonable to not assume flowsettings.py in kotaemon. And let the pipeline creator for ktem to decide the loader configurations for their pipeline in flowsettings.py
> 
> And that will depend on the settings of ktem and pipeline creator, which doesn't really relate to this PR

yes not related to this one, just a comment, hence I didn't make a review.

**Pull request merged**